cpp/contracts/adapter.cpp

```diff
@@ -46,16 +46,21 @@ void adapter::create(
 
    auto _token_bytes = token_bytes.extract_as_byte_array();
    check(_token_bytes.size() == 32, "token bytes length must be 32");
-   check(is_account(token), "token account does not exist");
+
+   // Do not check token validity if token is not EOS native
+   if (token != name(0)) check(is_account(token), "token account does not exist");
```

Comment:
Be careful that there's no native distinction on EOS: it's a token as the other ones, check the stats table on [eosio.token](https://bloks.io/account/eosio.token?loadContract=true&tab=Tables&table=stat&account=eosio.token&scope=EOS&limit=100)

In other words, if EOS is being wrapped `token` would be equal to `eosio.token`, which is a valid account

Reply:
Here I mean, do not check if the token is not native in a  EOS chain like. With name(0), which is address = '' I expect that the native token is from some other chain type (like EVM for exampel)

Reply:
Ah you're right, I missed that, sorry! 

The "native" word in the comment misled me, I think it's better to stick to our definition though, I started to call it "local" referring to the unwrapped version chain, because "native" goes in conflict with the xerc20 definition of native, which is the chain's currency